### PR TITLE
Progress Dialog and bug fix

### PR DESCRIPTION
**Pull request by AlessioVerardo**
In this PR, 
we add progress dialog almost everywhere where a call to the database is made. We also invert the position of argument in getEvents and getEventEdits method in the eventdatabase. Finally, comments about the last PR are taken into account. 

Do not hesitate to make any comment. This PR should the usability of our app. 
Cheers :)

### CODE REVIEW

File: app/src/main/java/com/github/sdpteam15/polyevents/model/entity/DeviceLocation.kt [hunk 1]
```diff
@@ -3,6 +3,12 @@ package com.github.sdpteam15.polyevents.model.entity
 import com.google.android.gms.maps.model.LatLng
 import java.time.LocalDateTime
 
+/**
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/entity/DeviceLocation.kt:6]: Nice adding comments

File: app/src/main/java/com/github/sdpteam15/polyevents/model/map/RouteMapHelper.kt [hunk 2]
```diff
@@ -624,9 +626,14 @@ object RouteMapHelper {
...
+        var dim_dp = 35
+        if(context != null){
+            dim_dp = dim_dp.dpToPixelsFloat(context).toInt()
+        }
+
         val anchor = IconAnchor(0.5f, 0.5f)
-        val bound = IconBound(0, 0, 100, 100)
-        val dimension = IconDimension(100, 100)
+        val bound = IconBound(0, 0, dim_dp, dim_dp)
+        val dimension = IconDimension(dim_dp, dim_dp)
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/map/RouteMapHelper.kt:636]: Nice feature

**Review by Mathematou — APPROVED** [review]
Nice adding the progress dialog, maybe we don't need to have it everywhere but we cannot know before trying

**Review by Rygone — APPROVED** [review]
good use of the Progress Dialog

**Pull request merged**